src/injector.ts

```diff
@@ -62,6 +73,12 @@ export class Injector {
     provide(token : any, defaultValue? : any): any;
     provide(token : any, defaultValue? : any): any {
         let hasDefault = arguments.length > 1;
+        // TODO: ask @rezonant if this is safe
+
+        // if it's a generic ref, lookup the specific match we put in the map
+        if( token instanceof ReflectedGenericRef && token.isGeneric()){
+          token = this.#generics.find(t => t.matchesRef(token)) || token
```

Comment:
Assuming `matchesRef(..)` works out, we can essentially resolve the token here with a simple list of generic tokens, then let things play out as normal :)